onnx/defs/logical/defs.cc

```diff
@@ -16,9 +18,10 @@ The result is a tensor of type integer in which `0` mean false and `1` mean true
         schema.NumInputs(2);
         schema.NumOutputs(1);
         schema.SetDoc(doc);
-        schema.Input(0, "left", "Left input tensor for the logical operator.");
-        schema.Input(1, "right", "Right input tensor for the logical operator.");
-        schema.Output(0, "output", "Result tensor of type `int`, 0 mean False and 1 mean True.");
+        schema.Input(0, "left", "Left input tensor for the logical operator.", "T");
+        schema.Input(1, "right", "Right input tensor for the logical operator.", "T");
+        schema.Output(0, "output", "Result tensor of type `int`, 0 mean False and 1 mean True.", "T");
+        schema.TypeConstraint("T", { "tensor(int32)" }, "Constrain input and output types to int tensor.");
```

Comment:
We don't really need to use a type-variable T if it is constrained to take a single value, do we? This looks fine too, but just want to check that it would be equivalent to saying "tensor(int32)" in-place without "T".

Reply:
Or, is the string/type parameter of Input/output always interpreted as a type-variable?

Reply:
The grammar specified in schema.h says that the parameter of Input/output is given by "<type> ::= <data_type> | tensor(<data_type>) | sparse(<data_type>) | <type_parameter>". Is there a corresponding parsing function? The function DataTypeUtils::FromString does not seem to consider the <type_parameter> case.

Reply:
One possible fix is to say that parameter of Input/output is always a type-variable and cannot be a type-constant like "tensor(float)"

Reply:
tensor(int32) should be put in-place. Fixed in new iteration.